### PR TITLE
Added @flaky to test_change_has_user

### PR DESCRIPTION
**Pull request by orangejenny**

### CODE REVIEW

**Review by emord — APPROVED** [review]
It might be nice to include the seen stack trace in either the github PR, or in a comment, so there's something to go off of if it gets fixed

**Comment by orangejenny** [comment]
@emord good point - I already restarted this job so the log is lost, but will include it in the future

File: corehq/apps/locations/tests/test_change_has_user.py [hunk 1]
```diff
@@ -66,6 +67,7 @@ def submit_form(self, name):
         self.assertTrue(form.is_valid())
         form.save()
 
+    @flaky
```
Comment (esoergel) [inline corehq/apps/locations/tests/test_change_has_user.py]: What does `@flaky` do?  Has this test been intermittently failing recently?

Comment (orangejenny) [reply corehq/apps/locations/tests/test_change_has_user.py]: By default, it re-attempts the test a second time and passes the build if that second attempt succeeds. https://pypi.org/project/flaky/

It's bitten me a few times in the past week. I don't know what our threshold should be for adding the decorator, open to suggestions.

Comment (esoergel) [reply corehq/apps/locations/tests/test_change_has_user.py]: huh, cool - I hadn't seen it before, looks like it was just added?  Seems like a reasonable approach.  I do hate essentially admitting defeat with flaky tests, but this seems vastly better than waiting for a failure and rebuliding the whole job, and better than requiring PR authors to fix unrelated test failures.

Comment (orangejenny) [reply corehq/apps/locations/tests/test_change_has_user.py]: I definitely share your ambivalence.

**Comment by orangejenny** [comment]
Tests are green, even though github is showing them still in progress. Merging.

**Comment by orangejenny** [comment]
Found an older PR where this failed:
```
corehq.apps.locations.tests.test_change_has_user:TestChangeHasUser.test ... FAIL
======================================================================
FAIL: corehq.apps.locations.tests.test_change_has_user:TestChangeHasUser.test
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/corehq/apps/locations/tests/test_change_has_user.py", line 81, in test
    self.submit_form('Essex')
  File "/mnt/commcare-hq-ro/corehq/apps/locations/tests/test_change_has_user.py", line 66, in submit_form
    self.assertTrue(form.is_valid())
AssertionError: False is not true
-------------------- >> begin captured logging << --------------------
casexml.apps.phone.restore_caching: DEBUG: invalidating ('RestorePayloadPathCache', u'test-has-user', u'', None, u'corehq.apps.users.models.CommCareUser.create_location_delegates')
casexml.apps.phone.restore_caching: DEBUG: invalidating ('RestorePayloadPathCache', u'test-has-user', u'', None, None)
celery.app.trace: INFO: Task corehq.messaging.tasks.sync_case_for_messaging[d2a6aaa5-9eae-4b28-b101-8c88aafaebe4] succeeded in 0.022450807s: None
casexml.apps.phone.restore_caching: DEBUG: invalidating ('RestorePayloadPathCache', u'test-has-user', u'', None, u'corehq.apps.users.models.CommCareUser.create_location_delegates')
casexml.apps.phone.restore_caching: DEBUG: invalidating ('RestorePayloadPathCache', u'test-has-user', u'', None, None)
celery.app.trace: INFO: Task corehq.messaging.tasks.sync_case_for_messaging[d5885f6a-6c9b-4365-ac45-36b4651874c2] succeeded in 0.018648064s: None
celery.app.trace: INFO: Task corehq.apps.locations.tasks.update_location_users[76f95736-8d78-46bf-a0d2-27fbb56cab7b] succeeded in 0.575573474s: None
--------------------- >> end captured logging << ---------------------
```

**Comment by orangejenny** [comment]
Hmm, looking more closely at what's failed, I think this is a legitimate failure. Will investigate and revert if it is.

**Pull request merged**